src/py21cmfast/src/SpinTemperatureBox.c

```diff
@@ -890,16 +893,38 @@ LOG_SUPER_DEBUG("got density gridpoints");
             }
 
             if (flag_options->USE_MINI_HALOS){
-                log10_Mcrit_mol = log10(lyman_werner_threshold(zp, 0.));
+                log10_Mcrit_mol = log10(lyman_werner_threshold(zp, 0., 0.,astro_params));
                 log10_Mcrit_LW_ave = 0.0;
-#pragma omp parallel shared(log10_Mcrit_LW_unfiltered,previous_spin_temp,zp) private(i,j,k) num_threads(user_params->N_THREADS)
+#pragma omp parallel shared(log10_Mcrit_LW_unfiltered,previous_spin_temp,zp) private(i,j,k,curr_vcb) num_threads(user_params->N_THREADS)
                 {
 #pragma omp for reduction(+:log10_Mcrit_LW_ave)
                     for (i=0; i<user_params->HII_DIM; i++){
                         for (j=0; j<user_params->HII_DIM; j++){
                             for (k=0; k<user_params->HII_DIM; k++){
-                                *((float *)log10_Mcrit_LW_unfiltered + HII_R_FFT_INDEX(i,j,k)) = \
-                                                log10(lyman_werner_threshold(zp, previous_spin_temp->J_21_LW_box[HII_R_INDEX(i,j,k)]));
+
+                              if (flag_options->FIX_VCB_AVG){ //with this flag we ignore reading vcb box
+                                curr_vcb = VAVG;
+                              }
+                              else{
+                                if(user_params->USE_RELATIVE_VELOCITIES){
+                                  curr_vcb = ini_boxes->lowres_vcb[HII_R_INDEX(i,j,k)];
+                                }
+                                else{ //set vcb to a constant, either zero or vavg.
+                                  curr_vcb = 0.0;
+                                }
+                              }
+
+                              *((float *)log10_Mcrit_LW_unfiltered + HII_R_FFT_INDEX(i,j,k)) = \
+                                              log10(lyman_werner_threshold(zp, previous_spin_temp->J_21_LW_box[HII_R_INDEX(i,j,k)],
+                                              curr_vcb, astro_params) );
+
+//JBM: There might be a better way than just reading ALL the ICs? Is the struct populated by defualt?
```

Comment:
it is on the TODO list (Issue #69 )

Reply:
Good, we'll circle back to this at some point then, thanks!